tarantool/connection.py

```diff
@@ -1089,11 +1105,16 @@ def replace(self, space_name, values, on_push=None, on_push_ctx=None):
             :exc:`~tarantool.error.DatabaseError`,
             :exc:`~tarantool.error.SchemaError`,
             :exc:`~tarantool.error.NetworkError`,
-            :exc:`~tarantool.error.SslError`
+            :exc:`~tarantool.error.SslError`,
+            :exc:`~tarantool.error.NotSupportedError`
 
         .. _replace: https://www.tarantool.io/en/doc/latest/reference/reference_lua/box_space/replace/
         """
 
+        if not self.fetch_schema:
```

Comment:
Btw, even though `load_schema` marked `:meta private:` in documentation, I think it's possible for one to load schema some time after even if they set `self.fetch_schema` to `false`. So
- let's check not the `fetch_schema` option, but the presence of `schema` itself,
- let's add a test that it's possible to call `load_schema`/`flush_schema` and make our connection schemafull again.

Reply:
Done, now the connection `fetch_schema` type can be changed. Also added tests with this case

Reply:
No, it shouldn't be like this. Changing internal structure parameter is not a good approach. I think you should simply
- do not fetch schema on connect if you wasn't asked to do so,
- do not allow to run requests using schema if there is not any schema,
- do not reload schema is there is none,
- allow to run requests using schema if there is one,
- reload schema is there is one.

Maybe there are some non-obvious points regarding schema clean-up between reconnects.

Reply:
Do I understand correctly:

* The `fetch_schema` attribute in the connection object should not be available to the user, it should be made private (now it is possible to change the parameter like `conn.fetch_schema=...` for an existing `conn` object, now `fetch_schema` is public). We want the `fetch_schema` parameter is manually set by user only when establishing a connection.
    + If it is decided to make the `fetch_schema` private, then we can implement a getter (`conn.is_fetch_schema_enable() -> Bool`), so we can understand the type of connection.
* However, the connection can be made from `schema-less` to `schema-full`. To do this, it is enough for the user to call `conn.update_schema(conn.schema_version)`. After that, all methods that were previously unavailable for `schema-less` connection become available. At the same time, the scheme starts updating automatically. And `conn.is_fetch_schema_enable()` will start returning `True`.
* It is not possible to change the connection type from `schema-full` to `schema-less`.

Reply:
> The fetch_schema attribute in the connection object should not be available to the user, it should be made private (now it is possible to change the parameter like conn.fetch_schema=... for an existing conn object, now fetch_schema is public)

There are no convenient in-built ways to declare a "private" in Python class: either you start it with underscore (and it still would be public) or mess with getters/setters overwriting, similar to Lua (and it's mostly don't worth it). You can leave it as is now, but never rely on someone would change that and do not teach others to do so through examples and tests.

> * If it is decided to make the `fetch_schema` private, then we can implement a getter (`conn.is_fetch_schema_enable() -> Bool`), so we can understand the type of connection.

It's also not worth it. I don't think that it be used that much often, and if one really needs it, he can just look up our "private" variable.



> * However, the connection can be made from `schema-less` to `schema-full`. To do this, it is enough for the user to call `conn.update_schema(conn.schema_version)`. After that, all methods that were previously unavailable for `schema-less` connection become available. At the same time, the scheme starts updating automatically. And `conn.is_fetch_schema_enable()` will start returning `True`.

> * It is not possible to change the connection type from `schema-full` to `schema-less`.

Yeah, I propose to use this approach.

Reply:
AFAIU, done. Now switching from `schema-less` to `schema-full` possible via `conn.update_schema(conn.schema_version)`, tests also cover this case

Reply:
Why we still base our conditions on parameter value rather that schema value itself?

Reply:
I don't think we should ever overwrite `fetch_schema` parameter which should be used to determine if schema would be fetched on connect or not and only there.

Reply:
Done

> Why we still base our conditional checks on parameter value rather that schema value itself?

Now the presence of the schema value is checked

> I don't think we should ever overwrite `fetch_schema` parameter which should be used to determine if schema would be fetched on connect or not and only there.

Now this value changes only inside `update_schema` from `False` to `True`